content/200-orm/200-prisma-client/500-deployment/301-edge/450-deploy-to-cloudflare.mdx

```diff
@@ -536,6 +545,193 @@ npx wrangler deploy
 
 The command will output the URL where you can access the deployed Worker.
 
-### D1
+### Cloudflare D1
+
+If you are using a D1 database, you need to:
+
+- use the `@prisma/adapter-d1` database adapter (via the `driverAdapters` Preview feature)
+- set `sqlite` as the `datasource` provider in your Prisma schema
+- manually generate SQL statements for schema changes using `prisma migrate diff` but execute them using [D1's migration system](https://developers.cloudflare.com/d1/reference/migrations/)
+
+You can find a [deployment-ready example on GitHub](https://github.com/prisma/prisma-examples/blob/latest/deployment-platforms/edge/cloudflare-workers/with-d1).
+
+#### 1. Configure Prisma schema
+
+> **Note**: If you don't have a project to deploy, follow the instructions in the [Prerequisites](#prerequisites) to bootstrap a basic Cloudflare Worker with Prisma ORM in it.
+
+In your Prisma schema, add the `driverAdapters` Preview feature to the `generator` block and set the `provider` of the `datasource` to `sqlite`. If you just bootstrapped the Prisma schema with `prisma init`, also be sure to add the following `User` model to it:
+
+```prisma file=schema.prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["driverAdapters"]
+}
+
+datasource db {
+  provider = "sqlite"
+  url      = env("DATABASE_URL")
+}
+
+model User {
+  id    Int     @id @default(autoincrement())
+  email String  @unique
+  name  String?
+}
+```
+
+Note that in this tutorial, you won't need the `.env` file since the connection between Prisma ORM and D1 will happen through a [binding](https://developers.cloudflare.com/workers/configuration/bindings/).
+
+#### 2. Install dependencies
+
+Next, install the required packages:
+
+```terminal
+npm install @prisma/adapter-d1
+```
+
+#### 3. Set the D1 database connection via a binding
+
+To connect your Workers with the D1 instance, add the following binding to your `wrangler.toml` (if you don't have a D1 instance yet, you can create one using the [Cloudflare Dashboard](https://dash.cloudflare.com/) or with the [`wrangler d1 create`](https://developers.cloudflare.com/workers/wrangler/commands/#create) command):
+
+```toml file=wrangler.toml
+name = "prisma-cloudflare-worker-example"
+main = "src/index.ts"
+compatibility_date = "2024-03-20"
+compatibility_flags = ["nodejs_compat"]
+
+[[d1_databases]]
+binding = "DB" # i.e. available in your Worker on env.DB
+database_name = "__YOUR_D1_DATABASE_NAME__" # to be replaced
+database_id = "__YOUR_D1_DATABASE_ID__"     # to be replaced
+```
+
+Note that `__YOUR_D1_DATABASE_NAME__` and `__YOUR_D1_DATABASE_ID__` in the snippet above are placeholders that should be replaced with the database name and ID of your own D1 instance. If you weren't able to grab this ID from the terminal output, you can also find it in the Cloudflare Dashboard or by running `npx wrangler d1 list` and `npx wrangler d1 info __YOUR_D1_DATABASE_NAME__` in your terminal.
```

Comment:
I think it'd be useful to add a code snippet / screenshot of:

```sh
❯ npx wrangler d1 create prisma-d1
 ⛅️ wrangler 3.39.0
-------------------
✅ Successfully created DB 'prisma-d1' in region EEUR
Created your database using D1's new storage backend. The new storage backend is not yet recommended for
production workloads, but backs up your data via point-in-time restore.

[[d1_databases]]
binding = "DB"
database_name = "prisma-d1"
database_id = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
```

This would clarify that in this example we call the binding `DB` (which is later used in the documentation step "Add the `DB` to the `Env` interface.", in section 5).